client.go

```diff
@@ -112,6 +113,12 @@ type CheckRetry func(resp *http.Response, err error) (bool, error)
 // that should pass before trying again.
 type Backoff func(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration
 
+// ErrorHandler is called if retries are expired, containing the last status
+// from the http library. The default will close the body and return an error
+// indicating how many tries were attempted. If overriding this, be sure to
+// close the body if needed.
+type ErrorHandler func(resp *http.Response, err error, numTries int) (*http.Response, error)
```

Comment:
My opinion is that we don't need this, and can return the response and error directly. That pattern isn't unprecedented, since net/http already does that for failed redirects. It also shouldn't break any existing code, since there is no guarantees made about the first value in the presence of an error. 

I'm not opposed to this addition if there a solid reason, but I need some convincing ;)

Reply:
Really, it's just for backwards compat. I'm not opposed to making that the default behavior if we just want to force others to get in line as I think it makes sense.